### PR TITLE
second batch of dropping deprecated APIs for 2.8

### PR DESCRIPTION
**Pull request by johanandren**
On top of #31866 but separate since massive, only akka-stream

References #31664

### CODE REVIEW

**Review by octonato — APPROVED** [review]
LGTM

Since 2.5.x? Impressive! 😄

**Comment by johanandren** [comment]
Biggest change is that we added a ton of Java Duration APIs back then and deprecated FiniteDuration in the Java APIs.

File: akka-stream/src/main/scala/akka/stream/Attributes.scala [hunk 1]
```diff
@@ -288,36 +288,10 @@ final class Attributes private[akka] (
...
-    getFirst[T] match {
-      case Some(a) => a
-      case None    => default
-    }
-  }
-
   /**
    * Scala API: Get the least specific attribute (added first) of a given type parameter T `Class` or subclass thereof.
    */
+  // deprecated but used by Akka HTTP so needs to stay
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/Attributes.scala:294]: 👍

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**